### PR TITLE
Instrument obsreport.Processor

### PR DESCRIPTION
**Pull request by moh-osman3**
Description:

This PR instruments obsreport.Processor with otel go. This is a followup PR that is based on similar changes made for obsreport.Receiver: https://github.com/open-telemetry/opentelemetry-collector/pull/6222

Link to tracking Issue: Part of https://github.com/open-telemetry/opentelemetry-collector/issues/816

Testing:
Ran contrib collector with memory_limiter processor and prometheusreceiver + sent traces from demo client and confirmed that new metrics show up. Tested both with feature gate disabled and enabled.

From output of $ curl localhost:8888/metrics

```
# HELP otelcol_processor_accepted_metric_points Number of metric points successfully pushed into the next component in the pipeline.
# TYPE otelcol_processor_accepted_metric_points counter
otelcol_processor_accepted_metric_points{processor="memory_limiter",service_instance_id="d957e470-f4a9-4448-b0b9-e2a2e98a2b53",service_name="otelcontribcol",service_version="v0.63.0-48-ge9099f1c57"} 298
# HELP otelcol_processor_accepted_spans Number of spans successfully pushed into the next component in the pipeline.
# TYPE otelcol_processor_accepted_spans counter
otelcol_processor_accepted_spans{processor="memory_limiter",service_instance_id="d957e470-f4a9-4448-b0b9-e2a2e98a2b53",service_name="otelcontribcol",service_version="v0.63.0-48-ge9099f1c57"} 84
# HELP otelcol_processor_dropped_metric_points Number of metric points that were dropped.
# TYPE otelcol_processor_dropped_metric_points counter
otelcol_processor_dropped_metric_points{processor="memory_limiter",service_instance_id="d957e470-f4a9-4448-b0b9-e2a2e98a2b53",service_name="otelcontribcol",service_version="v0.63.0-48-ge9099f1c57"} 0
# HELP otelcol_processor_dropped_spans Number of spans that were dropped.
# TYPE otelcol_processor_dropped_spans counter
otelcol_processor_dropped_spans{processor="memory_limiter",service_instance_id="d957e470-f4a9-4448-b0b9-e2a2e98a2b53",service_name="otelcontribcol",service_version="v0.63.0-48-ge9099f1c57"} 0
# HELP otelcol_processor_refused_metric_points Number of metric points that were rejected by the next component in the pipeline.
# TYPE otelcol_processor_refused_metric_points counter
otelcol_processor_refused_metric_points{processor="memory_limiter",service_instance_id="d957e470-f4a9-4448-b0b9-e2a2e98a2b53",service_name="otelcontribcol",service_version="v0.63.0-48-ge9099f1c57"} 0
# HELP otelcol_processor_refused_spans Number of spans that were rejected by the next component in the pipeline.
# TYPE otelcol_processor_refused_spans counter
otelcol_processor_refused_spans{processor="memory_limiter",service_instance_id="d957e470-f4a9-4448-b0b9-e2a2e98a2b53",service_name="otelcontribcol",service_version="v0.63.0-48-ge9099f1c57"} 0
```

### CODE REVIEW

File: .chloggen/obsreport-processor.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+# The name of the component, or a single word describing the area of concern, (e.g. otlpreceiver)
+component: obsreport
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: "Instrument `obsreport.Processor` metrics with otel-go"
+
+# One or more tracking issues or pull requests related to the change
+issues: [6607]
+
+# (Optional) One or more lines of additional information to render under the primary note.
```
Comment (jpkrohling) [inline .chloggen/obsreport-processor.yaml:13]: Can you remove this if it's unused?

**Review by jpkrohling — APPROVED** [review]

File: obsreport/obsreport_processor.go [hunk 2]
```diff
@@ -54,134 +84,228 @@ type ProcessorSettings struct {
...
 // NewProcessor creates a new Processor.
 func NewProcessor(cfg ProcessorSettings) (*Processor, error) {
-	return &Processor{
-		level:    cfg.ProcessorCreateSettings.MetricsLevel,
-		mutators: []tag.Mutator{tag.Upsert(obsmetrics.TagKeyProcessor, cfg.ProcessorID.String(), tag.WithTTL(tag.TTLNoPropagation))},
-	}, nil
+	return newProcessor(cfg, featuregate.GetRegistry())
+}
+
+// Deprecated: [v0.65.0] use NewProcessor.
```
Comment (jpkrohling) [inline obsreport/obsreport_processor.go:90]: This PR will probably land after we release 0.66.0 (happening now), so, this needs to be updated.

**Comment by jpkrohling** [comment]
@paivagustavo, are you able to review this one as well, as the author of the PR that this is using as reference?

File: .chloggen/obsreport-processor.yaml (hunk 1 above)
Comment (moh-osman3) [reply .chloggen/obsreport-processor.yaml]: Thanks for the review! Removed.

File: obsreport/obsreport_processor.go (hunk 2 above)
Comment (moh-osman3) [reply obsreport/obsreport_processor.go]: Thanks for pointing this out! Removed this deprecated function now.

File: obsreport/obsreport_processor.go [hunk 3]
```diff
@@ -54,134 +84,218 @@ type ProcessorSettings struct {
...
+	}
+
+	if err := proc.createOtelMetrics(); err != nil {
+		return nil, err
+	}
+
+	return proc, nil
+}
+
+func (por *Processor) createOtelMetrics() error {
```
Comment (paivagustavo) [inline obsreport/obsreport_processor.go:109]: Let's keep the same pattern of passing the configs instance as parameter here and creating the meter inside this function as [the previous prs](https://github.com/open-telemetry/opentelemetry-collector/pull/6460/files#diff-2864c8597e07e090876715c3acba587547f0f86254491f01cc00faa7be1c301fR107)

**Review by paivagustavo — APPROVED** [review]
LGTM! Thanks @moh-osman3, I've added one small note regarding consistency with other otel instrumentations.

Sorry for taking too long, I was away in vacation :)

File: obsreport/obsreport_processor.go (hunk 3 above)
Comment (moh-osman3) [reply obsreport/obsreport_processor.go]: Thanks for the review! Fixed!

**Pull request merged**